docs/providers/clickhouse.mdx

```diff
@@ -0,0 +1,68 @@
+---
+title: "ClickHouse"
+description: "Featureform supports [ClickHouse](https://clickhouse.com/) as an Offline Store."
+---
+
+## Implementation
+
+### Primary Sources
+
+#### Tables
+
+Table sources are used directly via a view. Featureform will never write to a primary source.
+
+### Transformation Sources
+
+SQL transformations are used to create a view. By default, those views are materialized and updated according to the schedule parameter. Deprecated transformations are converted to un-materialized views to save storage space.
+
+### Offline to Inference Store Materialization
+
+When a feature is registered, Featureform creates an internal transformation to get the newest value of every feature and its associated entity using a [ClickHouse ASOF JOIN](https://clickhouse.com/docs/en/sql-reference/statements/select/join#asof-join-usage). A Kubernetes job is then kicked off to sync this up with the Inference store.
+
+### Training Set Generation
+
+Every registered feature and label is associated with a view table. That view contains three columns, the entity, value, and timestamp. When a training set is registered, it is created as a materialized view via a JOIN on the corresponding label and feature views.
+
+## Configuration
+
+First we have to add a declarative ClickHouse configuration in Python.
+
+clickhouse\_config.py
```

Comment:
this line should be part of line 32; similar to [Redshift](https://github.com/featureform/featureform/blob/main/docs/providers/redshift.mdx#L32)